tests/test_gmp.py

```diff
@@ -26,6 +27,16 @@ def test_mpz_from_to_str(x):
     assert str(mx) == sx
     assert str(mnx) == snx
 
+@given(integers(min_value=0))
+@example('20a')
+def test_mpz_from_str(x):
+    try:
+        mpz(x, 10)
```

Comment:
No, it's wrong.  ``x`` - is an integer (int), not a string.  This will raise a TypeError.

I would rather suggest you to expand the test_digits_frombase().  For base <= 36 - you can reduce generated base value (say, to average ~ ``(base+2)/2``) and then again try the constructor with same string.  Either both int and mpz will raise a ValueError, or they should be equal.  For base > 36 - you can do similar with gmpy2.mpz.

Reply:
But when I use `mpz('20',10)`, its running result is the same as `mpz(20, 10)`. Why?

Reply:
> But when I use mpz('20',10), its running result is the same as mpz(20, 10). Why?

I don't know.  It should be the same.  And it definitely not the same in the master branch:
```pycon
>>> from gmp import mpz
>>> mpz('20', 10)
20
>>> mpz(20, 10)
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    mpz(20, 10)
    ~~~^^^^^^^^
TypeError: can't convert non-string with explicit base
```